What's new and readme: we are moving to Python 3.7

josephsl opened the pull request:
### Link to issue number:
- Fixes #9862 
Gives closure to #7105 

### Summary of the issue:
What's new and readme still mentions Python 2.7, and in case of what's new, changes were needed.

### Description of how this pull request fixes the issue:
Edits include mentioning Python 3.7 in both readme and what's new, as well as fixing things here and there in what's new/Threshold section. See the below section regarding work included in what's new and readme through this pull request and other Threshold work.

### Testing performed:
None

### Known issues with pull request:
Some parts are missing, such as filling in which issue introduced support for pyserial 3.4 and other things. What's new and readme are subject to change based on comments.

### Change log entry:
None

### Pull requests addressed by this work

Items with a checkmark are mentioned in what's new. Merge messages are ignored.

#### Speech refactor

- [x] New speech framework including callbacks, beeps, sounds, profile switches and prioritized queuing (#7599)

#### Python 3.7

- [x] nvwave.playWaveFile: async > asynchronous (#8647)
- [x] Update Developer Guide re. tab completion (PR #9935)
- [x] Update to pySerial 3.4 (#8815)
- [x] Remove functionality marked as deprecated (#9603)
- [x] Update dependencies for Python 3 (#9630)
- [x] Bump NVDA version for threshold to 2019.3.0 and also set backwards compat to the same. We already know there will be 2019.2 before threshold, but this may need to be bumpt to 2019.4.0 if we end up doing a 2019.3 off master as well.
- [x] Compile liblouis with 32 bit widechars (#9544)
- [x] textUtils module to deal with offset differences between Python 3 strings and Windows wide character strings with surrogate characters (PR #9545)
- [x] [No entry necessary] - Python 3: Update py2exe to fix several issues (PR #9856)
- [x] Update UIAutomationCore.dll comtypes module to version 10.0.18362 (PR #9829)
- [x] Bump threshold version and add-on compatibility version to 2019.4.0 for now as master is already heading toward a 2019.3.
- [x] No longer require pyWin32 (#9639)
- [x] Appveyor: use Python 3.7 rather than Python 2.

## No entry necessary
- [x] Fixes for BrailliantB and FreedomScientific (PR #9839)
- [x] Python 2/3: relative imports (#9670)
- [x] Python 3: print statement -> print function (#9664)
- [x] Python 3: file function -> open function (#9663)
- [x] Python 3: use floor division (//) to obtain integer results from division operations (#9692)
- [x] Python 3: use importlib for importing modules (#9693)
- [x] Python 3: dictionary iteration/list methods made compatible with Python 3 (#9671)
- [x] Python 3: xrange -> range (#9627)
- [x] SAPI 5 synth driver: percentage to pitch/rate must return an integer. Re #9641. (#9699)
- [x] Python 3: except exception, e -> except as e (#9698)
- [x] Python 3: dict.haskey -> key in dict (#9697)
- [x] Check pot: file -> open (#9718)
- [x] WinConsole handler/Python 3: use floor divisoin assignment operator. Re #9641. (#9737)
- [x] Unittests/Python 3: update syntax, catch unbound method registration test in extension points as expected failure (#9726)
- [x] Python 3: zip function -> list(zip), itertools.izip -> zip() (#9729)
- [x] Global commands: restore commands affected by division operator differences between Python 2 and 3 (#9748)
- [x] Python 3: more native imports and modules (#9751)
- [x] Input core/input gestures dialog: script function object has no attribute __func__. (#9752)
- [x] Python 3: use native metaclass definition syntax (#9753)
- [x] System tests/Python 3: syntax and imports (#9801)
- [x] App module handler/does app mod exists: return false when called before imports are initialized  (#9800)
- [x] Add-on handler: ugettext -> gettext (#9810)
- [x] Python 3/urllib: imports, timeout (#9820)
- [x] Update check/Python 3: update check resource is encoded as bytes, thus make it Unicode. Re #9819. (#9837)
- [x] WinKernel: create a wrapper for MoveFileExW function and associated flag constants (PR #9849)
- [x] Python 3: use UTF-8 decoding for font name with Scintilla controls (PR #9893)
- [x] Python 3:  MoveFileEx - catch WindowsError instead of WinError (PR #9876)
- [x] Python 3: Use built-in "input" instread of "raw_input" (PR #9908)
- [x] Python 3: Check for None during check for previous installation version in Installer GUI (PR #9924)
- [x] Update setup script for Python 3.7 (#9605)
- [x] Update Scons build environment to run on Python 3 (#9667)
- [x] Python 3: Fix use of subprocess (#9716)
- [x] Python 3: No longer encode or decode if it is not necessary (#9734)
- [x] Python 3: fix several remaining division errors (#9774)
- [x] Write syntax errors and tracebacks to python console (#9767)
- [x] Python 3: fix logging level not being updated correctly (#9809)
- [x] Python 3: Fix deprecation warning extensionPoints.callWithSUpportedKwargs (PR #9778)
- [x] Fix comparison error in start menu (PR #9846)
- [x] Update BRLTTY driver (PR #9821)
- [x] Python 3 one core: Don't check for sys.frozen in the check method. (PR #9857)
- [x] Python 3: fix say all for DisplayModelEditableText (PR #9866)
- [x] Python 3: sys.exec_prefix > sys.prefix for nvda_slave (PR #9871)
- [x] Python 3: Fix usages of "open" function (PR #9853)
- [x] Python3: Fix save scripts in lockapp (PR #9884)
- [x] Python 3: When processing xml coming from virtual buffers, make sure that surrogate characters are handled properly (PR #9897)
- [x] Python 3: Remove many instances of "py3 review required" comments (PR #9890)
- [x] Python 3: Fix adobe reader int/str to NoneType comparisons as well as other identified cases (PR: #9881)
- [x] Revert "Restore VoiceSettingsSlider to ensure we don't break compatibility with Audio Themes add-on" (PR #9907)
- [x] Python 3: Clean the current words buffer in more cases (PR #9877)
- [x] Python 3: Remove Python 2 specific workarounds (PR #9889)
- [x] Python 3: Fix loading of espeak variants (PR #9920)
- [x] Python 3: Fix several local overrides of range function (PR #9912)
- [x] Fix waiting for outlook/powerpoint window to exist for ever (PR #9950)
- [x] Log never handled speech indexes, and don't make the speech queue lag behind (PR #9946)
- [x] Fix error in JAVA text ranges (PR #9951)
- [x] appveyor: build snapshots for threshold
- [x] OneCore synthDriver: fix up merge conflicts.
- [x] Only clear curWordChars on focus / caret movement (#9597)
- [x] import Python 3 native modules (#9703)
- [x] Python 3: sys.maxint -> sys.maxsize, as maxint is gone in Python 2. Re #7105. (#9704)
- [x] Python3: os.getcwdu -> os.getcwd (#9705)
- [x] Python 3/remote python console: SocketServer -> socketserver. (#9706)
- [x] Update miscDeps to latest master, containing python3 txt2tags.
- [x] Python 3: the thread module is no longer available. Use threading instead. (#9713)
- [x] Python3: fix remaining syntax errors found by py2exe (#9719)
- [x] Python3: fix NVDA's logging infrastructure (#9712)
- [x] Python3: gettext is unicode, and pgettext is placed in builtins properly (#9714)
- [x] objidl: HRESULT is no longer available in ctypes.wintypes. Get it from comtypes. (#9745)
- [x] remove use of unichr, basestring and unicode objects. (#9724)
- [x] Remove need for hashing NVDAObject and TextInfo objects (#9746)
- [x] Revert "Remove need for hashing NVDAObject and TextInfo objects (#9746)" (#9756)
- [x] Python3: base virtualBuffer: coerce attribute values to a string before escaping for regular expression  (#9759)
- [x] cmp no longer exists in Python3. Replace with (a>b)-(a<b) (#9758)
- [x] Python3: Provide __hash__ methods where ever we have provided __eq__ methods (#9757)
- [x] Correct a few places where we are trying to do integer comparisons with None. (#9794)
- [x] Ensure NVDA contains a signed and valid python37.dll (#9796)
- [x] Python3: implement new way of converting COMError to CallCancelled (#9795)
- [x] bdDetect: change basestring to str. This was the only thing causing NVDA not to run in this branch, and will be introduced exactly the same by pr #9736 anyway.
- [x] Make output of both py2exe and nose more verbose to aide in future build issues.
- [x] Revert "App module handler/does app mod exists: return false when called before imports are initialized  (#9800)" (#9815)
- [x] config.addConfigDirsToPythonPackagePath: When adding scratchpad dirs, create them if they don't already exist. (#9814)
- [x] Update miscDeps to contain new brlApi module (#9812)
- [x] Python3: stop using unichr in brailleInput.py (#9835)
- [x] Python3 convert braille drivers (PR #9736)
- [x] Fix buildversion for threshold branch
- [x] Fix api back compat version number
- [x] Python 3: Fix missing import, remove unused imports (PR #9901)
- [x] Python 3: Fix error with accPropServer (PR #9900)
- [x] txt2tags for python3 #8734 (#9648)
- [x] Change iter.next() to next(iter) (PR #9535)
- [x] Remove .pyc and .pyo files when updating (PR #9960)
- [x] Python3: Fix SAPI5 and MSSP on installed NVDA (PR #9964)
- [x] Detect caret movement using events, except for IA2Web #9933

josephsl committed:
What's new: remove extraneous indentation.

josephsl committed:
What's new: add missing punctuation.

josephsl committed:
What's new: say hello to Python 3.7! Re #7105

josephsl commented:
Hi all,

Comments from everyone who has worked on or commented on Python 3 are appreciated, especially from Reef, @JCSTeh, and @leonardder.

Thanks,

XLTechie commented on user_docs/en/changes.t2t:
```diff
@@ -10,12 +10,12 @@ What's New in NVDA
-- config.getConfigDirs has been removed
-- config.ConfigManager.getConfigValidationParameter is removed. A bug in getConfigValidation has been fixed in order for it being able to fully replace getConfigValidationParameter
+- config.getConfigDirs has been removed.
+- config.ConfigManager.getConfigValidationParameter is removed. A bug in getConfigValidation has been fixed in order for it being able to fully replace getConfigValidationParameter.
```
Can this line possibly be made more clear? Something like this, perhaps?

```suggestion
- config.ConfigManager.getConfigValidationParameter has been removed. A bug in getConfigValidation was fixed in order for it to be able to fully replace getConfigValidationParameter.
```

codeofdusk replied:
Even better, you can remove the words "in order" from that sentence.

josephsl commented:
Hi, right, thanks for catching this, which also had a grammar problem (will resolve it once I wake up from sleep in the morning). Thanks. 

 

From: Luke Davis <notifications@github.com> 
Sent: Tuesday, July 16, 2019 8:54 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] What's new and readme: we are moving to Python 3.7 (#9942)

 

@XLTechie commented on this pull request.

  _____  

In user_docs/en/changes.t2t <https://github.com/nvaccess/nvda/pull/9942#discussion_r304210746> :

>  - braille.BrailleHandler._get/set_tether have been removed. use getTether and setTether respectively.
 
-- config.getConfigDirs has been removed
 
-- config.ConfigManager.getConfigValidationParameter is removed. A bug in getConfigValidation has been fixed in order for it being able to fully replace getConfigValidationParameter
 
+- config.getConfigDirs has been removed.
 
+- config.ConfigManager.getConfigValidationParameter is removed. A bug in getConfigValidation has been fixed in order for it being able to fully replace getConfigValidationParameter.
 

Can this line possibly be made more clear? Something like this, perhaps?

⬇️ Suggested change 

-- config.ConfigManager.getConfigValidationParameter is removed. A bug in getConfigValidation has been fixed in order for it being able to fully replace getConfigValidationParameter.
 
+- config.ConfigManager.getConfigValidationParameter has been removed. A bug in getConfigValidation was fixed in order for it to be able to fully replace getConfigValidationParameter.
 

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/9942?email_source=notifications&email_token=AB4AXEAPYBSAPKGVOECQQ63P72JXDA5CNFSM4IEJ26ZKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB6U555A#pullrequestreview-262790900> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AB4AXEFWHX2M5QO3Z4URRB3P72JXDANCNFSM4IEJ26ZA> .

leonardder commented on readme.md:
```diff
@@ -35,7 +35,7 @@ The NVDA source depends on several other packages to run correctly.
 The following dependencies need to be installed on your system:
 
-* [Python](http://www.python.org/), version 2.7.15, 32 bit
+* [Python](http://www.python.org/), version 3.7.4, 32 bit
```
Please consider #8421

Note that on appveyor, we can't ensure that we are on a particular minor release. Therefore, I propose that we advice to always use the most recent minor version

```suggestion
* [Python](http://www.python.org/), most recent minor release of version 3.7, 32 bit
```

leonardder commented on user_docs/en/changes.t2t:
```diff
@@ -10,12 +10,13 @@ What's New in NVDA
 
 == Changes for Developers ==
- - Updated pySerial to version 3.4
+- Updated Python to 3.7.4. (#7105)
```
Same as above.

leonardder commented on user_docs/en/changes.t2t:
```diff
@@ -10,12 +10,13 @@ What's New in NVDA
 - the validate module is now only available from configobj. Code should now do from configobj import validate rather than import validate.
-- textInfos.Point and textInfos.Rect are removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively 
+- textInfos.Point and textInfos.Rect are removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively.
 - braille.BrailleHandler._get/set_tether have been removed. use getTether and setTether respectively.
```
```suggestion
- braille.BrailleHandler._get/set_tether have been removed. Use getTether and setTether respectively.
```

LeonarddeR reviewed (changes requested):
You can change the threshold release heading to 2019.3

josephsl commented:
Hi, I knew there was an issue about mentioning minor versions in docs – thanks for the link to the issue. Thanks.

 

From: Leonard de Ruijter <notifications@github.com> 
Sent: Tuesday, July 16, 2019 10:21 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] What's new and readme: we are moving to Python 3.7 (#9942)

 

@leonardder requested changes on this pull request.

You can change the threshold release heading to 2019.3

  _____  

In readme.md <https://github.com/nvaccess/nvda/pull/9942#discussion_r304222505> :

> @@ -35,7 +35,7 @@ The NVDA source depends on several other packages to run correctly.
 
 ### Installed Dependencies
 
 The following dependencies need to be installed on your system:
 
 
 
-* [Python](http://www.python.org/), version 2.7.15, 32 bit
 
+* [Python](http://www.python.org/), version 3.7.4, 32 bit
 

Please consider #8421 <https://github.com/nvaccess/nvda/issues/8421> 

Note that on appveyor, we can't ensure that we are on a particular minor release. Therefore, I propose that we advice to always use the most recent minor version

⬇️ Suggested change 

-* [Python](http://www.python.org/), version 3.7.4, 32 bit
 
+* [Python](http://www.python.org/), most recent minor release of version 3.7, 32 bit
 
  _____  


In user_docs/en/changes.t2t <https://github.com/nvaccess/nvda/pull/9942#discussion_r304223038> :

> @@ -10,12 +10,13 @@ What's New in NVDA
 
 
 
 
 
 == Changes for Developers ==
 
- - Updated pySerial to version 3.4
 
+- Updated Python to 3.7.4. (#7105)
 

Same as above.

  _____  

In user_docs/en/changes.t2t <https://github.com/nvaccess/nvda/pull/9942#discussion_r304223116> :

>  - the validate module is now only available from configobj. Code should now do from configobj import validate rather than import validate.
 
-- textInfos.Point and textInfos.Rect are removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively 
 
+- textInfos.Point and textInfos.Rect are removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively.
 
 - braille.BrailleHandler._get/set_tether have been removed. use getTether and setTether respectively.
 

⬇️ Suggested change 

- - braille.BrailleHandler._get/set_tether have been removed. use getTether and setTether respectively.
 
+- braille.BrailleHandler._get/set_tether have been removed. Use getTether and setTether respectively.
 

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/9942?email_source=notifications&email_token=AB4AXEDALRUZTDZNP3VRROTP72T4RA5CNFSM4IEJ26ZKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB6VBMQA#pullrequestreview-262805056> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AB4AXEHRKSBAC56DLTQNLPLP72T4RANCNFSM4IEJ26ZA> .

josephsl committed:
What's new: various gramar fixes.

josephsl committed:
What's new and readme: Python 3.7.4 -> 3.7. Re #8421.

Because Appveyor changes Python x.y.z fromtime to time, it is bettre to just say Python x.y.

josephsl committed:
What's new: more grammar fixes.

josephsl commented:
Hi,

Should we mention textUtils?

Thanks.

leonardder commented:
yes, sounds like a good idea. Probably best to point at examples from the changelog.

josephsl commented:
Hi,

Regarding text utils, I have nothing to say - can't come up with a descriptive changelog for developers at the moment.

Anything else from our PR pool? If not, I'll switch the PR to review mode.

Thanks.

leonardder commented:
How about something within the lines of:

* Added a new textUtils module that makes it easier to deal with string differences between Python 3 strings and Windows unicode strings. See the module documentation and textInfos.offsets module for example implementations. (#9545)

josephsl committed:
What's new: add textUtils entry. Re #9545.

What's new entry suggested by Leonard de Ruijter (Babbage).

josephsl commented:
Hi,

Text utils: done.

Thanks.

josephsl committed:
What's new: grammar fixes. Re #9918.

feerrenrut reviewed (approved):
This is looking pretty good. Please update the description to cover which PR's (to threshold / threshold staging) have been covered by this. You can get most of the way there by copying entries from git log on the threshold branch. I suggest using the checkbox markdown:

```
- [ ] some PR not covered
- [x] some PR covered
```

josephsl commented:
Hi,

I'll use git shortlog to edit the description.

Thanks.

josephsl commented:
Hi,

To @LeonardDer: whenever you are ready, please take a look at this PR. Thanks.

feerrenrut commented:
Thanks @josephsl, I have updated the description so that they render as a checkboxes. It looks like there are still many PR's that need consideration.

josephsl commented:
Hi, yes, including the just merged one (#9960). Thanks.

 

From: Reef Turner <notifications@github.com> 
Sent: Tuesday, July 23, 2019 5:25 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] What's new and readme: we are moving to Python 3.7 (#9942)

 

Thanks @josephsl <https://github.com/josephsl> , I have updated the description so that they render as a checkboxes. It looks like there are still many PR's that need consideration.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/9942?email_source=notifications&email_token=AB4AXEG3HISXLWS2VO5I2C3QA32DJA5CNFSM4IEJ26ZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2S577Q#issuecomment-514187262> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AB4AXEBEMAPZZ25TPQ755NLQA32DJANCNFSM4IEJ26ZA> .

feerrenrut commented:
If no one volunteers before tomorrow (25-July 0600 UTC+0) , then I'll finish this off.

leonardder commented on user_docs/en/changes.t2t:
```diff
@@ -3,31 +3,33 @@ What's New in NVDA
-- config.getConfigDirs has been removed
-- config.ConfigManager.getConfigValidationParameter is removed. A bug in getConfigValidation has been fixed in order for it being able to fully replace getConfigValidationParameter
-- inputCore.InputGesture.logIdentifier property has been removed. Now use inputCore.InputGesture.identifiers[0] instead.
+- textInfos.Point and textInfos.Rect has been removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively.
```
```suggestion
- textInfos.Point and textInfos.Rect have been removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively.
```

leonardder commented on user_docs/en/changes.t2t:
```diff
@@ -3,31 +3,33 @@ What's New in NVDA
-- config.ConfigManager.getConfigValidationParameter is removed. A bug in getConfigValidation has been fixed in order for it being able to fully replace getConfigValidationParameter
-- inputCore.InputGesture.logIdentifier property has been removed. Now use inputCore.InputGesture.identifiers[0] instead.
+- textInfos.Point and textInfos.Rect has been removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively.
+- braille.BrailleHandler._get/set_tether has been removed. Use getTether and setTether respectively.
```
```suggestion
- braille.BrailleHandler._get/set_tether have been removed. Use getTether and setTether respectively.
```

LeonarddeR reviewed (dismissed):
I found two sentences where has should be have IMO. Could be wrong for the latter, though.

josephsl commented:
Hi, I think part of that may have to do with subtle English grammar differences. 

 

From: Leonard de Ruijter <notifications@github.com> 
Sent: Wednesday, July 24, 2019 4:41 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] What's new and readme: we are moving to Python 3.7 (#9942)

 

@leonardder requested changes on this pull request.

I found two sentences where has should be have IMO. Could be wrong for the latter, though.

  _____  

In user_docs/en/changes.t2t <https://github.com/nvaccess/nvda/pull/9942#discussion_r306763881> :

>  - the validate module is now only available from configobj. Code should now do from configobj import validate rather than import validate.
 
-- textInfos.Point and textInfos.Rect are removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively 
 
-- braille.BrailleHandler._get/set_tether have been removed. use getTether and setTether respectively.
 
-- config.getConfigDirs has been removed
 
-- config.ConfigManager.getConfigValidationParameter is removed. A bug in getConfigValidation has been fixed in order for it being able to fully replace getConfigValidationParameter
 
-- inputCore.InputGesture.logIdentifier property has been removed. Now use inputCore.InputGesture.identifiers[0] instead.
 
+- textInfos.Point and textInfos.Rect has been removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively.
 

⬇️ Suggested change 

-- textInfos.Point and textInfos.Rect has been removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively.
 
+- textInfos.Point and textInfos.Rect have been removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively.
 
  _____  


In user_docs/en/changes.t2t <https://github.com/nvaccess/nvda/pull/9942#discussion_r306764104> :

>  - the validate module is now only available from configobj. Code should now do from configobj import validate rather than import validate.
 
-- textInfos.Point and textInfos.Rect are removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively 
 
-- braille.BrailleHandler._get/set_tether have been removed. use getTether and setTether respectively.
 
-- config.getConfigDirs has been removed
 
-- config.ConfigManager.getConfigValidationParameter is removed. A bug in getConfigValidation has been fixed in order for it being able to fully replace getConfigValidationParameter
 
-- inputCore.InputGesture.logIdentifier property has been removed. Now use inputCore.InputGesture.identifiers[0] instead.
 
+- textInfos.Point and textInfos.Rect has been removed and replaced by locationHelper.Point and locationHelper.RectLTRB, respectively.
 
+- braille.BrailleHandler._get/set_tether has been removed. Use getTether and setTether respectively.
 

⬇️ Suggested change 

-- braille.BrailleHandler._get/set_tether has been removed. Use getTether and setTether respectively.
 
+- braille.BrailleHandler._get/set_tether have been removed. Use getTether and setTether respectively.
 

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/9942?email_source=notifications&email_token=AB4AXEDVT2H5WHDIMDWZ4XLQBA5TNA5CNFSM4IEJ26ZKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB7ND4BQ#pullrequestreview-265960966> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AB4AXECKC3WCHZQKGGEYY3DQBA5TNANCNFSM4IEJ26ZA> .

josephsl commented:
Hi,

Let me at least commit the requested edits, and then I think we should decide our next steps.

Thanks.

josephsl committed:
What's new: address grammar edit comments.

Reviewed by Leonard de Ruijter (Babbage): has -> have for two entries.

feerrenrut commented:
I'd like to get this finished off and merged in, are you going to continue with it @josephsl? I don't think I will hold back threshold any longer.

josephsl commented:
Hi, feel free to merge – let’s get the party started. Thanks.

 

From: Reef Turner <notifications@github.com> 
Sent: Thursday, July 25, 2019 7:05 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] What's new and readme: we are moving to Python 3.7 (#9942)

 

I'd like to get this finished off and merged in, are you going to continue with it @josephsl <https://github.com/josephsl> ? I don't think I will hold back threshold any longer.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/9942?email_source=notifications&email_token=AB4AXECCPTEEJ7VQXFZHM7TQBGXHFA5CNFSM4IEJ26ZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2ZSJ3A#issuecomment-515056876> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AB4AXEABL7XCIMYUWNDHW6TQBGXHFANCNFSM4IEJ26ZA> .

josephsl commented:
Hi,

Clarification: I'm done with this one.

Thanks.

leonardder commented:
@josephsl: could you please do a base branch switch for this?

josephsl commented:
Done.

josephsl committed:
Merged master

derekriemer commented:
Should this be merged?

josephsl commented:
Hi, when it is ready, yes. Thanks.

 

From: Derek Riemer <notifications@github.com> 
Sent: Sunday, August 4, 2019 2:48 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] What's new and readme: we are moving to Python 3.7 (#9942)

 

Should this be merged?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/9942?email_source=notifications&email_token=AB4AXEEONXHMPTK5O3BXH2LQC5FATA5CNFSM4IEJ26ZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD3QKLZI#issuecomment-518039013> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AB4AXEHQU6C5UBK2GBGFXBLQC5FATANCNFSM4IEJ26ZA> .

feerrenrut commented:
@derekriemer The checklist for PR's needs to be completed, I suspect many of these do not need an entry. This is on my list of things to do, but hasn't yet received priority.

feerrenrut committed:
Merge remote-tracking branch 'origin/master' into HEAD

feerrenrut committed:
Remaining what's new changes for threshold

feerrenrut commented:
I have gone through the unchecked items and added entries.

I also re-arranged the current items in the changes file.
